src/thor/triplegbuilder.cc

```diff
@@ -462,6 +462,287 @@ void SetHeadings(TripLeg_Edge* trip_edge,
+  if (controller.attributes.at(kNodeIntersectingEdgeWalkability)) {
+    intersecting_edge->set_walkability(GetTripLegTraversability(traversability));
+  }
+
+  traversability = Traversability::kNone;
+  // Determine cyclability
+  if (intersecting_de->forwardaccess() & kBicycleAccess) {
+    traversability = (intersecting_de->reverseaccess() & kBicycleAccess) ? Traversability::kBoth
+                                                                         : Traversability::kForward;
+  } else {
+    traversability = (intersecting_de->reverseaccess() & kBicycleAccess) ? Traversability::kBackward
+                                                                         : Traversability::kNone;
+  }
+  // Set the cyclability flag for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeCyclability)) {
+    intersecting_edge->set_cyclability(GetTripLegTraversability(traversability));
+  }
+
+  // Set the driveability flag for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeDriveability)) {
+    intersecting_edge->set_driveability(
+        GetTripLegTraversability(nodeinfo->local_driveability(local_edge_index)));
+  }
+
+  // Set the previous/intersecting edge name consistency if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeFromEdgeNameConsistency)) {
+    bool name_consistency =
+        (prev_de == nullptr) ? false : prev_de->name_consistency(local_edge_index);
+    intersecting_edge->set_prev_name_consistency(name_consistency);
+  }
+
+  // Set the current/intersecting edge name consistency if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeToEdgeNameConsistency)) {
+    intersecting_edge->set_curr_name_consistency(directededge->name_consistency(local_edge_index));
+  }
+
+  // Set the use for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeUse)) {
+    intersecting_edge->set_use(GetTripLegUse(intersecting_de->use()));
+  }
+
+  // Set the road class for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeRoadClass)) {
+    intersecting_edge->set_road_class(GetRoadClass(intersecting_de->classification()));
+  }
+
+  // Set the lane count for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeLaneCount)) {
+    intersecting_edge->set_lane_count(intersecting_de->lanecount());
+  }
+
+  // Set the sign info for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeSignInfo)) {
+    if (intersecting_de->sign()) {
+      // Need the index of the intersecting-edge to access its sign info from the tile
+      GraphId endnode = intersecting_de->endnode();
+      valhalla::baldr::graph_tile_ptr t2 = graphreader.GetGraphTile(endnode);
+      const DirectedEdge* zeroth_de = t2->directededge(0);
+      size_t idx = intersecting_de - zeroth_de;
```

Comment:
Is there a better way to get the index/offset of a directed-edge?  This works fine but seems a little hokey.

Reply:
you shouldnt use the end node to get the tile and reference that with the intersecting edge. the intersecting edge is in the same tile as its begin node. its end node may be in another tile.

Reply:
you can probably just use the tile that was passed in and do the same thing you were doing. nothign wrong with pointer arithmetic.

```c++
auto edge_index =  intersecting_de - graphtile->directededge(0);
```

Reply:
@kevinkreiser I think I've fixed this let me know thx.